staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go

```diff
@@ -180,6 +177,33 @@ func (p *Patcher) patchSimple(obj runtime.Object, modified []byte, source, names
 	return patch, patchedObj, err
 }
 
+// getPatchType returns correct PatchType for the given GVK.
+// It fetches supported patch types directly from OpenAPISchema, if it is initiated.
+// Otherwise, it uses conventional registration method to determine patch type.
+func (p *Patcher) getPatchType(gvk schema.GroupVersionKind) (types.PatchType, error) {
+	if p.OpenapiSchema != nil {
```

Comment:
Why would this be nil?

Reply:
If `openapi-patch` flag https://github.com/kubernetes/kubernetes/blob/8d427a2ddcc909d1a3ed1a3c1ec8475952603b1d/staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go#L228 is explicitly set to false.

OpenapiSchema is nil because of this https://github.com/kubernetes/kubernetes/blob/f493336f7758bab25c33ff96f1f98e7bf3fa52c5/staging/src/k8s.io/kubectl/pkg/cmd/apply/patcher.go#L78

I don't know `openapi-patch` flag is used?

Reply:
Also, I see the flag as being set to true by default (both in the code, and when I run kubectl).